assume/units/storage.py

```diff
@@ -368,11 +372,17 @@ def calculate_min_max_discharge(
         )
 
         # restrict according to min_soc
-        max_soc_discharge = self.calculate_soc_max_discharge(
-            self.outputs["soc"].at[start]
-        )
+        if soc is None:
+            soc = self.outputs["soc"].at[start]
+        max_soc_discharge = self.calculate_soc_max_discharge(soc)
         max_power_discharge = max_power_discharge.clip(max=max_soc_discharge)
 
+        # convert to single values if only one value is left
+        if len(min_power_discharge) == 1:
+            min_power_discharge = min_power_discharge[0]
+        if len(max_power_discharge) == 1:
+            max_power_discharge = max_power_discharge[0]
+
```

Comment:
this does not give a clear return type - requiring to test this if the return is float or array..?

Even though I understand it is tempting, I would rather keep this to return a list with one element in these cases..?